api/notd/gallery_manager.py

```diff
@@ -176,28 +176,32 @@ async def query_collection_tokens(self, registryAddress: str, limit: int, offset
             .where(orderedListingsQuery.c.tokenListingIndex == 1)
         ).subquery()
         listingsQuery = lowestListingsQuery #if usesListings else LatestTokenListingsTable
+        ownerQuery =(
+            sqlalchemy.select(TokenOwnershipsTable)
+            .where(TokenOwnershipsTable.c.registryAddress == registryAddress)
+        ).subquery()
         if collection.doesSupportErc721:
             query = (
                 sqlalchemy.select(TokenMetadatasTable, TokenCustomizationsTable, listingsQuery, sqlalchemy.literal(1).label('quantity'))
                     .join(TokenCustomizationsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == TokenCustomizationsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == TokenCustomizationsTable.c.tokenId), isouter=True)
                     .join(TokenOwnershipsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == TokenOwnershipsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == TokenOwnershipsTable.c.tokenId))
                     .join(listingsQuery, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == listingsQuery.c.registryAddress, TokenMetadatasTable.c.tokenId == listingsQuery.c.tokenId, listingsQuery.c.offererAddress == TokenOwnershipsTable.c.ownerAddress), isouter=True)
+                    .join(ownerQuery, sqlalchemy.and_(ownerQuery.c.registryAddress == listingsQuery.c.registryAddress, ownerQuery.c.tokenId == listingsQuery.c.tokenId, ownerQuery.c.ownerAddress == listingsQuery.c.offererAddress), isouter=True)
                     .where(TokenMetadatasTable.c.registryAddress == registryAddress)
                     .order_by(sqlalchemy.cast(TokenMetadatasTable.c.tokenId, sqlalchemy.Integer).asc())
                     .limit(limit)
                     .offset(offset)
             )
         elif collection.doesSupportErc1155:
             query = (
-                sqlalchemy.select(TokenMetadatasTable, TokenCustomizationsTable, sqlalchemy.func.sum(TokenMultiOwnershipsTable.c.quantity).label('quantity'))
```

Comment:
if this stuff doesnt work my suggestion would be undo this for now (so we dont return listings and we can add it in a separate PR